src/StoreKeeper/WooCommerce/B2C/Backoffice/MetaBoxes/ProductSyncMetaBox.php

```diff
@@ -0,0 +1,156 @@
+        if ($storekeeperId && StoreKeeperOptions::isConnected()) {
+            $backofficeLabel = esc_html__('Open in backoffice', I18N::DOMAIN);
+            $backofficeLink = esc_attr(StoreKeeperOptions::getBackofficeUrl()."#products/details/$storekeeperId");
+            $backoffice = <<<HTML
+                        <a href="$backofficeLink" class="product_backoffice_link" target="_blank">$backofficeLabel</a>
+                        HTML;
+        }
+
+        $submitLabel = esc_html__('Force sync', I18N::DOMAIN);
+        echo <<<HTML
+            <ul class="product_actions submitbox">
+                <li class="wide">
+                    <div>   
+                        <strong>$idLabel:</strong>
+                        <div>$idValue</div>
+                    </div>
+                    <div>
+                        <strong>$dateLabel:</strong>
+                        <div>$dateValue</div>
+                    </div>
+                </li>
+                <li class="wide">        
+                    <a href="$syncUrl" class="button-primary product_sync_submission">$submitLabel</a>
+                    $backoffice
+                </li>
+            </ul>
+            <style>
+                #poststuff #storekeeper-product-sync .inside {
+                    margin: 0;
+                    padding: 0;
+                } 
+                #poststuff #storekeeper-product-sync .inside ul.product_actions li {
+                    padding: 6px 10px;
+                    box-sizing: border-box;
+                } 
+                #poststuff #storekeeper-product-sync .inside .product_sync_submission {
+                    float: right;
+                } 
+            </style>
+            HTML;
+
+        $this->showPossibleError();
+        $this->showPossibleSuccess();
+    }
+
+    /**
+     * Function to sync product from storekeeper backoffice to wordpress.
+     *
+     * @throws \Exception
+     */
+    final public function doSync(int $postId): void
+    {
+        $nonce = array_key_exists('_wpnonce', $_REQUEST) ? $_REQUEST['_wpnonce'] : null; // no need to escape, wp_verify_nonce compares hash
+        if (1 === wp_verify_nonce($nonce, self::ACTION_NAME.'_post_'.$postId)) {
+            if (wc_get_product($postId)) {
+                $product = wc_get_product($postId);
+                $storekeeperId = $this->getPostMeta($product->get_id(), 'storekeeper_id', false);
+                $tasks = $this->getTasks($storekeeperId);
+
+                if (empty($tasks)) {
+                    $noUpdateMessage = __('Product is in sync and no updates were found.', I18N::DOMAIN);
+                    wp_redirect(
+                        get_edit_post_link($postId, 'url').'&sk_sync_success='.urlencode($noUpdateMessage)
+                    );
+                    exit();
+                }
+
+                $processor = new ProcessSingleTask();
+                try {
+                    foreach ($tasks as $task) {
```

Comment:
synching once is enough

Reply:
For loop is now for updating tasks statuses only